sources/tools/Stride.Importer.Assimp/MeshConverter.cs

```diff
@@ -0,0 +1,1364 @@
+
+                // TODO TEMP: Set a default diffuse model
+                finalMaterial.Attributes.DiffuseModel = new MaterialDiffuseLambertModelFeature();
+            }
+            else if (textureType == TextureType.TextureTypeSpecular)
+            {
+                var specularFeature = new MaterialSpecularMapFeature
+                {
+                    SpecularMap = computeColorNode
+                };
+                finalMaterial.Attributes.Specular = specularFeature;
+
+                // TODO TEMP: Set a default specular model
+                var specularModel = new MaterialSpecularMicrofacetModelFeature
+                {
+                    Fresnel = new MaterialSpecularMicrofacetFresnelSchlick(),
+                    Visibility = new MaterialSpecularMicrofacetVisibilityImplicit(),
+                    NormalDistribution = new MaterialSpecularMicrofacetNormalDistributionBlinnPhong()
+                };
+                finalMaterial.Attributes.SpecularModel = specularModel;
+            }
+            else if (textureType == TextureType.TextureTypeEmissive)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeAmbient)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeReflection)
+            {
+                // TODO: Add support
+            }
+            if (textureType == TextureType.TextureTypeOpacity)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeShininess)
+            {
+                // TODO: Add support
+            }
+            if (textureType == TextureType.TextureTypeSpecular)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeNormals)
+            {
+                finalMaterial.Attributes.Surface = new MaterialNormalMapFeature(computeColorNode);
+            }
+            else if (textureType == TextureType.TextureTypeDisplacement)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeHeight)
+            {
+                // TODO: Add support
+            }
+        }
+
+        private unsafe IComputeColor GenerateOneTextureTypeLayers(Silk.NET.Assimp.Material* pMat, TextureType textureType, int textureCount, MaterialAsset finalMaterial)
+        {
+            var stack = Material.Materials.ConvertAssimpStackCppToCs(assimp, pMat, textureType);
+
+            var compositionFathers = new Stack<IComputeColor>();
+
+            var sets = new Stack<int>();
+            sets.Push(0);
+
+            var nbTextures = assimp.GetMaterialTextureCount(pMat, textureType);
+
+            IComputeColor curComposition = null, newCompositionFather = null;
+
+            var isRootElement = true;
```

Comment:
So if i understood well, a material element is root if it's not currently referencing the previous element of the stack, or it's not being referenced by the next element of the stack.
```suggestion
            //var isRootElement = true; Should go in the loop
            
            bool isRootElement = top.type != Material.StackType.Operation && stack.Peek().type != Material.StackType.Operation;
            
```